src/rust/engine/tar_api/src/tar_api.rs

```diff
@@ -0,0 +1,95 @@
+extern crate flate2;
+extern crate tar;
+#[cfg(test)]
+extern crate tempfile;
+#[cfg(test)]
+extern crate testutil;
+
+use flate2::read::GzDecoder;
+use std::fs::File;
+use std::path::Path;
+use tar::Archive;
+
+pub fn decompress_tgz(tar_path: &Path, output_dir: &Path) -> Result<(), std::io::Error> {
+  let tar_gz = File::open(tar_path)?;
+  let tar = GzDecoder::new(tar_gz);
+  let mut archive = Archive::new(tar);
+  archive.unpack(output_dir)?;
+  Ok(())
+}
+
+#[cfg(test)]
+pub mod tar_tests {
+  use super::decompress_tgz;
+  use flate2::write::GzEncoder;
+  use flate2::Compression;
+  use std::fs::File;
+  use std::path::{Path, PathBuf};
+  use tempfile::TempDir;
+  use testutil::file::contents;
+  use testutil::make_file;
+
+  #[test]
+  fn decompress_normal_tar_file() {
+    // prepare a file containing 'hello world'
+    let tmp_dir = TempDir::new().unwrap();
+    let content = "hello world".as_bytes().to_vec();
+    let txt_full_path = std::fs::canonicalize(tmp_dir.path())
+      .unwrap()
+      .join(&PathBuf::from("hello.txt"));
+    make_file(&txt_full_path, &content, 0o600);
+
+    let path_in_tar = "a/b/c/d.txt";
+    let tgz_path = std::fs::canonicalize(tmp_dir.path())
+      .unwrap()
+      .join(&"simple.tgz");
+
+    let compress_result = compress(&txt_full_path, &path_in_tar, &tgz_path);
```

Comment:
I would write this as:
```rust
compress(&txt_full_path, &path_in_tar, &tgz_path).expect("Error compressing");
```

because then it will output the error in the failure message; just asserting on a boolean won't show that

Reply:
Much better pattern. Thanks!